packages/x-components/src/x-modules/queries-preview/components/__tests__/query-preview.spec.ts

```diff
@@ -0,0 +1,236 @@
+      query: 'milk',
+      rows: 24
+    });
+
+    await updateExtraParams({ store: 'Uganda' });
+
+    expect(eventSpy).toHaveBeenNthCalledWith(3, {
+      extraParams: { store: 'Uganda' },
+      origin: 'popular_search:none',
+      query: 'milk',
+      rows: 24
+    });
+  });
+
+  it('renders the results names in the default slot', () => {
+    const { queryPreview, findTestDataById } = renderQueryPreview();
+
+    const wrappers = findTestDataById('result-name');
+
+    queryPreview!.results.forEach((result, index) => {
+      expect(wrappers.at(index).element).toHaveTextContent(result.name!);
+    });
+  });
+
+  it('renders the specified number of results', () => {
+    const maxItemsToRender = 2;
+    const { findTestDataById } = renderQueryPreview({
+      maxItemsToRender
+    });
+
+    expect(findTestDataById('result-name')).toHaveLength(maxItemsToRender);
+  });
+
+  it('exposes the query, the results and the totalResults in the default slot', () => {
+    const template = `
+      <QueryPreview
+          :query="query"
+          #default="{ results, query, totalResults}">
+        <div>
+          <span data-test="query-preview-query">{{ query }}</span>
+          <span data-test="total-results">{{ totalResults }}</span>
+          <div v-for="result in results" :key="result.id">
+            <span data-test="result-name">{{result.name}}</span>
+          </div>
+        </div>
+      </QueryPreview>`;
+
+    const { query, wrapper, queryPreview, findTestDataById } = renderQueryPreview({
+      template
+    });
+
+    expect(wrapper.find(getDataTestSelector('query-preview-query')).element).toHaveTextContent(
+      query
+    );
+    expect(wrapper.find(getDataTestSelector('total-results')).element).toHaveTextContent(
+      queryPreview!.totalResults.toString()
+    );
+
+    const resultsWrappers = findTestDataById('result-name');
+
+    queryPreview!.results.forEach((result, index) => {
+      expect(resultsWrappers.at(index).element).toHaveTextContent(result.name!);
+    });
+  });
+
+  it('allows changing the result content', () => {
+    const template = `
+      <QueryPreview :query="query" #result="{ result }">
+        <span data-test="result-content">{{result.id}} - {{result.name}}</span>
+      </QueryPreview>
+    `;
+    const { findTestDataById, queryPreview } = renderQueryPreview({ template });
+
+    const resultsWrapper = findTestDataById('result-content');
+
+    queryPreview!.results.forEach((result, index) => {
+      expect(resultsWrapper.at(index).element).toHaveTextContent(`${result.id} - ${result.name!}`);
+    });
+  });
+
+  it('wont render if there are no results', () => {
+    const { wrapper } = renderQueryPreview({
+      queryPreview: {
+        request: {
+          query: 'milk'
+        },
+        results: [],
+        status: 'initial',
+        totalResults: 0
+      }
+    });
+
+    expect(wrapper.find(getDataTestSelector('query-preview')).element).toBeFalsy();
```

Comment:
method `find` is deprecated. in this case you can use `findComponent(QueryPreview)`

https://v1.test-utils.vuejs.org/api/wrapper/find.html

Reply:
Good point, I have changed it. Actually, `renderQueryPreview` is already using `findComponent` when returning the `wrapper` 🤦‍♂️